src/types/utility-types/enumeration.ts

```diff
@@ -8,14 +8,11 @@ export type UnionStringArray<T extends readonly string[]> = T[number]
 // these overloads also allow both mutable and immutable arrays, making types.enumeration<Enum>(Object.values(Enum)) possible.
 // the only case where this doesn't work is when passing to the function an array variable with a mutable type constraint;
 // for these cases, it will just fallback and assume the type is a generic string.
-export function enumeration<T extends readonly string[]>(
-  options: T
-): ISimpleType<UnionStringArray<T>>
```

Comment:
I was kind of surprised that this made the snapshot/creation types of `types.enumeration(["a", "b", "c"])` `string` instead of `"a" | "b" | "c"`. This change aligns it with the other overloaded signature.

My main concern is that, although I think this is the correct thing to do, users may have been relying on the wrong types. If they did provide a wrong value at runtime it would have failed, but if they weren't doing that they may find themselves having to do a lot more `as "a" | "b" | "c"`.

Reply:
Ah yeah. This is a tough call. This will personally break some of my own application code, but I think it's much more correct.

We spoke about this on Discord, but I'm willing to at least merge this in for now and ship it as a pre-release version. Maybe we move more slowly with this one than we have with prior pre-releases and get feedback.

Alternatively, we could ship the minor version, and revert this in a patch very quickly.

It's tough to balance stability with improving TypeScript, since most TypeScript changes can look like breaking changes. 

We won't rush on a release of this, but I like the change a lot, so I will make sure it's consumable via npm for you and anyone else who wants it. Thank you very much!

Reply:
Yeah, it's a tough call to make!

The other thing I forgot to mention is that this makes it consistent with the named overload too. It'd likely be confusing to people to have these be different:

```
types.enumeration("Letter", ["a", "b"])  // "a" | "b"
types.enumeration(["a", "b"])            // string
```

but I'm guessing it's rare that people move from the latter to the former, in which case they'd encounter a similar issue outlined in my first comment.

Reply:
Hmm, here's what I'm seeing on the main branch:

![CleanShot 2024-02-26 at 09 14 00](https://github.com/mobxjs/mobx-state-tree/assets/833576/86a40aac-1660-49a2-86da-0ceb796ae1ad)

And in my branch:

![CleanShot 2024-02-26 at 09 16 39](https://github.com/mobxjs/mobx-state-tree/assets/833576/4d9e7b09-120d-4320-894b-395e3d63f15a)

So it's definitely possible you have a configuration difference, or maybe a slightly older version of TS?

Reply:
Ah yeah, I see this in the MST repo as well. My application codebase is on TS 5.3.3, but there must be some kind of config problem on my end. I'll investigate and probably [update the TS tips page, which is very outdated](https://mobx-state-tree.js.org/tips/typescript).

I'm going to resolve this discussion and point to it in the preview release notes where I'll solicit feedback on breaking/not breaking version numbers. Thanks!